### PR TITLE
Add flight checkbox on activity add/edit form

### PR DESCRIPTION
**Pull request by ananyeet**
### What is a quick description of the change?

Add a "this is a flight" checkbox on the activity add form to make for less clicking.

### Is this fixing an issue?

none

### Are there more details that are relevant?

This PR is one of many created in an effort to reduce the number of button clicks a user makes.

### Check lists (check `x` in `[ ]` of list items)

- [  ] Test written/updating
- [  ] Tests passing
- [x] Coding style (indentation, etc)

Please explain why any are not present, if any.
This PR does not change any functions which need to be tested.

### Any additional comments?

Before:
 
image

After:
SLURP (4)

### CODE REVIEW

**Comment by zghera** [comment]
Could you add before and after pictures please?

File: frontend/src/components/ViewActivities/activitylist.js [hunk 1]
```diff
@@ -20,18 +20,18 @@ export async function getActivityList(tripId) {
...
-      data[DB.ACTIVITIES_START_TIME] =
-        data[DB.ACTIVITIES_START_TIME]['seconds'] * 1000;         
-      data[DB.ACTIVITIES_END_TIME] = 
-        data[DB.ACTIVITIES_END_TIME]['seconds'] * 1000;
+        if (data[DB.ACTIVITIES_END_TIME] !== undefined) {
+          // Eliminate nanoseconds, convert to milliseconds.
+          data[DB.ACTIVITIES_START_TIME] =
+            data[DB.ACTIVITIES_START_TIME]['seconds'] * 1000;         
+          data[DB.ACTIVITIES_END_TIME] = 
+            data[DB.ACTIVITIES_END_TIME]['seconds'] * 1000;
```
Comment (zghera) [inline frontend/src/components/ViewActivities/activitylist.js:33]: Optional nit: make 'seconds' into a constant in `database.js`

**Comment by ananyeet** [comment]
> Could you add before and after pictures please?

Added. Sorry!

File: frontend/src/components/ViewActivities/editActivity.js [hunk 2]
```diff
@@ -58,11 +66,20 @@ class EditActivity extends React.Component {
 
     newVals[DB.ACTIVITIES_START_COUNTRY] = 
       getRefValue(this.editStartLocRef, 'No Change', activity[DB.ACTIVITIES_START_COUNTRY]);
-    newVals[DB.ACTIVITIES_END_COUNTRY] = 
-      getRefValue(this.editEndLocRef, 'No Change', activity[DB.ACTIVITIES_END_COUNTRY]);
+    
+    // Ref doesn't keep track of checked value?
```
Comment (zghera) [inline frontend/src/components/ViewActivities/editActivity.js:70]: If this is true based on what you observed, remove '?' for clarity sake

File: frontend/src/components/ViewActivities/editActivity.js [hunk 3]
```diff
@@ -58,11 +66,20 @@ class EditActivity extends React.Component {
...
+    newVals[DB.ACTIVITIES_FLIGHT] = this.state.flightCheck;
     
     newVals[DB.ACTIVITIES_START_TZ] = getRefValue(this.editStartTzRef);
-    newVals[DB.ACTIVITIES_END_TZ] = getRefValue(this.editEndTzRef);
+
+    if (newVals[DB.ACTIVITIES_FLIGHT]) { // This is a flight.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/editActivity.js:75]: This comment and the one on the else are unnecessary in my opinion.

File: frontend/src/components/ViewActivities/editActivityFormElements.js [hunk 4]
```diff
@@ -62,18 +61,45 @@ export function locationElementFormGroup(controlId, formLabel, dropdown) {
...
     </Form.Group>
   );
 }
+
+/**
+ * Create a From Group with a checkbox.
+ * 
+ * @param {string} controlId FormGroup's control ID. 
+ * @param {string} formLabel Label of the field for this FormGroup. 
+ * @param {ref} ref The input's reference.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/editActivityFormElements.js:86]: I am not sure if I mentioned this before. But for refs, I think the correct type is `React.RefObject`

File: frontend/src/components/ViewActivities/index.js [hunk 5]
```diff
@@ -92,21 +97,26 @@ class ViewActivities extends React.Component {
...
           </div>
         );
       } else {
         return (
-          <Modal.Dialog>
-            <Modal.Header>Add New Activity</Modal.Header>
+          <Modal 
+          show={this.state.addingActivity}
+          onHide={this.cancelAdd}
+          dialogClassName="add-activity-modal" >
```
Comment (zghera) [inline frontend/src/components/ViewActivities/index.js:111]: Optional nit. It is most important to be consistent within your own code but I think Keiffer and I are using this syntax for element definitions like this.
```suggestion
          <Modal 
            show={this.state.addingActivity}
            onHide={this.cancelAdd}
            dialogClassName="add-activity-modal" 
          >
```

File: frontend/src/components/ViewActivities/index.js [hunk 6]
```diff
@@ -92,21 +97,26 @@ class ViewActivities extends React.Component {
       if (!this.state.addingActivity) {
         return (
           <div>
+            <Button type='button' onClick={this.addActivity}>+ Add</Button>
```
Comment (zghera) [inline frontend/src/components/ViewActivities/index.js:100]: Nit: Is "+ Add" redundant? If not then no change but if you think so consider removing the "+" or "Add".

File: frontend/src/components/ViewActivities/editActivity.js (hunk 3 above)
Comment (ananyeet) [reply frontend/src/components/ViewActivities/editActivity.js:75]: I do prefer them, makes things clearer.

**Comment by ananyeet** [comment]
Received @Hiram-Silvey 's verbal approval to merge since this is the last day of the internship and he was not able to review the PR.

**Pull request merged**